docs/src/examples/numpy.rst

```diff
@@ -0,0 +1,94 @@
+.. _numpy:
+
+Conversion to NumPy and Other Frameworks
+========================================
+
+MLX array implements the `Python Buffer Protocol <https://docs.python.org/3/c-api/buffer.html>`_.
+Let's convert an array to NumPy and back.
+
+.. code-block:: python
+
+  import mlx.core as mx
+  import numpy as np
+
+  a = mx.arange(3)
+  b = np.array(a)
+  c = mx.array(b)
```

Comment:
This will always be a copy, perhaps it could be mentioned.

Reply:
it is mentioned below in "By default, NumPy copies data to a new array." but I also added an in place comment for clarification.